app/scripts/modules/core/src/serverGroup/serverGroupSearchResultType.tsx

```diff
@@ -46,22 +54,127 @@ const SearchResultsHeader: SearchResultsHeaderComponent = () => (
     <HeaderCell col={cols.ACCOUNT}/>
     <HeaderCell col={cols.REGION}/>
     <HeaderCell col={cols.EMAIL}/>
+    <HeaderCell col={cols.HEALTH}/>
   </TableHeader>
 );
 
-const SearchResultsData: SearchResultsDataComponent = ({ results }) => (
+const SearchResultsData: SearchResultsDataComponent<IServerGroupSearchResult> = ({ results }) => (
   <TableBody>
-    {results.slice().sort(itemSortFn).map(item => (
+    {results.map(item => (
       <TableRow key={itemKeyFn(item)}>
         <HrefCell item={item} col={cols.SERVERGROUP} />
         <AccountCell item={item} col={cols.ACCOUNT} />
         <BasicCell item={item} col={cols.REGION} />
         <BasicCell item={item} col={cols.EMAIL} />
+        <HealthCountsCell item={item} col={cols.HEALTH} />
       </TableRow>
     ))}
   </TableBody>
 );
 
+interface IServerGroupDataProps {
+  type: ISearchResultType;
+  results: IServerGroupSearchResult[],
+}
+
+interface IServerGroupDataState {
+  serverGroups: IServerGroupSearchResult[],
+}
+
+interface IServerGroupTuple {
+  toFetch: IServerGroupSearchResult;
+  fetched: IServerGroup;
+}
+
+const makeServerGroupTuples = (sgToFetch: IServerGroupSearchResult[], fetched: IServerGroup[]): IServerGroupTuple[] => {
+  const findFetchedValue = (toFetch: IServerGroupSearchResult) => fetched.find(sg => sg.name === toFetch.serverGroup);
+  return sgToFetch.map(toFetch => ({ toFetch, fetched: findFetchedValue(toFetch) }));
+};
+
+const fetchServerGroups = (toFetch: IServerGroupSearchResult[]): Observable<IServerGroupTuple[]> => {
+  const fetchPromise = ReactInjector.API.one('serverGroups')
+    .withParams({ serverGroupNames: toFetch.map(sg => `${sg.account}:${sg.region}:${sg.serverGroup}`) })
+    .get()
+    .then((fetched: IServerGroup[]) => makeServerGroupTuples(toFetch, fetched));
+
+  return Observable.fromPromise(fetchPromise);
+};
+
+/**
+ * HOC which provides provides server group health/instance counts
+ *
+ * The /search endpoint does not return instance counts.
+ * This component fetches instance counts separately and provides them to the wrapped component.
+ * This component waits until it is rendered, then starts fetching instance counts.
+ * It mutates the input search results and then passes the data to the nested component.
+ */
+const AddHealthCounts = (Component: SearchResultsDataComponent<IServerGroupSearchResult>): SearchResultsDataComponent<IServerGroupSearchResult> => {
+  return class FetchHealthCounts extends React.Component<IServerGroupDataProps, IServerGroupDataState> {
+    public state = { serverGroups: [] } as any;
+    private results$ = new BehaviorSubject<IServerGroupSearchResult[]>([]);
+    private stop$ = new Subject();
+
+    constructor(props: any) {
+      super(props);
+
+      // results to use when a fetch has failed.
+      const failedFetch = (failedFetches: IServerGroupSearchResult[]) =>
+        Observable.of(failedFetches.map(toFetch => ({ toFetch, fetched: { instanceCounts: null } as IServerGroup })));
```

Comment:
Would be nice to use a more explicit flag to indicate fetch status instead of relying on `null` being an error and `undefined` being loading.

(It's not clear without reading the code that the two values represent significantly different states)

Reply:
I didn't update this part but I updated the other parts